### ISSUE TITLE
Annotation for regular Marks

### ISSUE THREAD

**Issue by smeikx** [chentoast/marks.nvim#78]
It seems annotations only work for bookmarks. Is there any way of adding annotations to native marks as well?
Also, could these annotations persist between sessions? Would it be possible to use shada for this?

**Comment by chentoast** [comment]
Yes, this is possible, but difficult to do and thus low priority. Extmarks can have extra tagged data associated with them (like annotations), but regular marks are much more inflexible. The solution would be to implement regular mark tracking with extmarks and just sync extmark positions to the mark positions every so often, but that would be a big refactor.
